### PR TITLE
FEM-2158: Fix excessive amount of TrackInfoRealm objects

### PR DESCRIPTION
**Pull request by noamtamim**
This fix includes a major refactor in our Realm handling.

### CODE REVIEW

File: Sources/DB/DTGItemRealm.swift [hunk 1]
```diff
@@ -15,21 +15,27 @@ import RealmSwift
...
+        self.selected = selected
+        
     }
     
+    
     public override class func shouldIncludeInDefaultSchema() -> Bool { return false } 
 
     func asTrackInfo() -> TrackInfo {
-        return TrackInfo(languageCode: self.languageCode, title: self.title)
+        return TrackInfo(languageCode: self.languageCode, title: self.title, type: TrackInfo.TrackType(rawValue: type)!)
```
Comment (x-NR-x) [inline Sources/DB/DTGItemRealm.swift]: Let's solve this in a different way, without '!'.
Don't want it to crash by any chance.

File: Sources/DB/DB.swift [hunk 2]
```diff
@@ -12,15 +12,53 @@
...
+        guard let newObj = newObj, let oldObj = oldObj else {return}
+
+        let names = [
+            (TrackInfo.TrackType.text, "selectedTextTracks", "textTracks"), 
+            (TrackInfo.TrackType.audio, "selectedAudioTracks", "audioTracks")
+        ]
+        
+        for (type, oldName, newName) in names {
+
+            let oldTracks = oldObj[oldName] as! List<DynamicObject>
```
Comment (x-NR-x) [inline Sources/DB/DB.swift]: Let's try to solve this in a different way, without '!'.
Don't want it to crash by any chance.

File: Sources/DB/DB.swift [hunk 3]
```diff
@@ -12,15 +12,53 @@
...
+            let newTracks = List<TrackInfoRealm>()
+            
+            for t in oldTracks {
+                let ti = TrackInfo(languageCode: t["languageCode"] as! String, title: t["title"] as! String, type: type)
```
Comment (x-NR-x) [inline Sources/DB/DB.swift]: Let's try to solve this in a different way, without '!'.
Don't want it to crash by any chance.

File: Sources/DB/DB.swift [hunk 4]
```diff
@@ -12,15 +12,53 @@
...
+                let tir = TrackInfoRealm(itemId: oldObj["id"] as! String, selected: true, trackInfo: ti)
```
Comment (x-NR-x) [inline Sources/DB/DB.swift]: Let's try to solve this in a different way, without '!'.
Don't want it to crash by any chance.

**Comment by noamtamim** [comment]
Not crashing is a good thing... but in all of those cases, there's nothing to do except to crash. 

- The migration code will not crash because all of the assumptions it makes **must** be valid during the migration from schema v2 to v3.
- The asTrackInfo() code will not crash because the string stored in db is always the rawValue of the enum.

In all of those cases, I can't think of any valid error handling path except to crash. Any other way will silently lose data.

**Comment by x-NR-x** [comment]
I think it's better to lose some data and download again.
And not that the app will not launch.

File: Sources/DB/DB.swift (hunk 2 above)
Comment (noamtamim) [reply Sources/DB/DB.swift]: In this case, it's actually better to crash because the app is in an illegal state.

**Pull request merged**